samples/ui-kit-place-search-nearby/index.ts

```diff
@@ -6,132 +6,137 @@
-            }).then(addMarkers);
-            // Handle user selection in Place Details.
-            placeList.addEventListener('gmp-placeselect', ({ place }) => {
-                markers[place.id].click();
-            });
+        if (place.location) {
+            map.innerMap.setCenter(place.location)
         }
-    });
-    /* [END maps_ui_kit_place_search_nearby_event] */
+    })
+    placeSearch.addEventListener(
+        'gmp-load',
+        () => {
+            searchPlaces()
+        },
+        { once: true }
+    )
 }
-
-async function addMarkers(){
-    const { AdvancedMarkerElement } = await google.maps.importLibrary('marker') as google.maps.MarkerLibrary;
-    const { LatLngBounds } = await google.maps.importLibrary('core') as google.maps.CoreLibrary;
-
-    const bounds = new LatLngBounds();
-
-    if(placeList.places.length > 0){
-        placeList.places.forEach((place) => {
-            let marker = new AdvancedMarkerElement({
-                map: map.innerMap,
-                position: place.location
-            });
-
-            markers[place.id] = marker;
-            bounds.extend(place.location);
-
-            /* [START maps_ui_kit_place_search_nearby_click_event] */
-            marker.addListener('gmp-click', (event) => {
-                if(infoWindow.isOpen){
-                    infoWindow.close();
-                }
-
-                placeDetailsRequest.place = place.id;
-                placeDetails.style.display = 'block';
-                placeDetails.style.width = '350px';
-                infoWindow.setOptions({
-                    content: placeDetails,
-                });
-                infoWindow.open({
-                    anchor: marker,
-                    map: map.innerMap
-                });
-            });
-            /* [END maps_ui_kit_place_search_nearby_click_event] */
-
-            map.innerMap.setCenter(bounds.getCenter());
-            map.innerMap.fitBounds(bounds);
-        });
+/* [END maps_ui_kit_place_search_nearby_event] */
+// The searchPlaces function is called when the user changes the type select or when the page loads.
+function searchPlaces() {
+    // Close the info window and clear the markers.
+    infoWindow.close()
+    for (const marker of markers.values()) {
+        marker.map = null
+    }
+    markers.clear()
+
+    // Set the place search query and add an event listener to the place search element.
+    if (typeSelect.value) {
+        const center = map.innerMap.getCenter()!
+        placeSearchQuery.maxResultCount = 5
+        placeSearchQuery.locationRestriction = {
+            center: { lat: center.lat(), lng: center.lng() },
+            radius: 50000, // 50km radius
+        }
+        placeSearchQuery.locationBias = {
+            center: { lat: center.lat(), lng: center.lng() },
+        }
+        placeSearchQuery.includedTypes = [typeSelect.value]
+        placeSearch.addEventListener('gmp-load', addMarkers, { once: true })
```

Comment:
consider moving this listener into the `init()`  (and then no "once:true" needed